pkg/controllers/csi/provisioner/controller.go

```diff
@@ -53,28 +49,20 @@ type imageInstallerBuilder func(context.Context, afero.Fs, *image.Properties) (i
 
 // OneAgentProvisioner reconciles a DynaKube object
 type OneAgentProvisioner struct {
-	client    client.Client
 	apiReader client.Reader
 	fs        afero.Fs
-	recorder  record.EventRecorder
-	db        metadata.Access
 
 	dynatraceClientBuilder dynatraceclient.Builder
 	urlInstallerBuilder    urlInstallerBuilder
 	imageInstallerBuilder  imageInstallerBuilder
-	opts                   dtcsi.CSIOptions
 	path                   metadata.PathResolver
 }
 
 // NewOneAgentProvisioner returns a new OneAgentProvisioner
-func NewOneAgentProvisioner(mgr manager.Manager, opts dtcsi.CSIOptions, db metadata.Access) *OneAgentProvisioner {
+func NewOneAgentProvisioner(mgr manager.Manager, opts dtcsi.CSIOptions) *OneAgentProvisioner {
 	return &OneAgentProvisioner{
-		client:                 mgr.GetClient(),
 		apiReader:              mgr.GetAPIReader(),
-		opts:                   opts,
 		fs:                     afero.NewOsFs(),
-		recorder:               mgr.GetEventRecorderFor("OneAgentProvisioner"),
```

Comment:
why do we completely remove the events? shouldn't we try to make them more useful or at least keep the same ones instead?

Reply:
IMO they are half-baked as of right now, I don't believe they are documented in any way, I have yet to see them being referenced in an RFA (or anywhere)

I would remove them for simplicity's sake, and add them back with more thought put into it.
- This can be done in context of the database removal